test/core.jl

```diff
@@ -106,6 +107,7 @@ A = @inferred(AxisArray(1:3, .1:.1:.3))
 @test axisnames(A) == (:row,)
 @inferred(axisnames(A))
 @test axisvalues(A) == (.1:.1:.3,)
+# FIXME: reintroduce inferred
 A = @inferred(AxisArray(reshape(1:16, 2,2,2,2), .5:.5:1))
```

Comment:
same as above. type stability breaks with last tuple element for `defaul_axes`

```
julia> @code_warntype AxisArrays.default_axes(reshape(1:16, 2,2,2,2), (.5:.5:1,))
Body::NTuple{4,Axis}
172 1 ─ %1  = (Base.getfield)(args, 1, true)::StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}    │╻╷          _default_axes
    │   %2  = %new(Axis{:row,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}, %1)::Axis{:row,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}
    │   %3  = (Base.getfield)(A, :dims)::NTuple{4,Int64}                                                                         │││╻╷╷╷        axes
    │   %4  = (Base.getfield)(%3, 1, true)::Int64                                                                                ││││╻           axes
    │   %22 = (getfield)(%18, 2)::Int64                                                                                          │││││││╻           tail

    [...]

    │         (Base.ifelse)(%30, 0, %24)                                                                                         ││││││││││┃           max
    │   %32 = %new(Axis{:page,Base.OneTo{Int64}}, %29)::Axis{:page,Base.OneTo{Int64}}                                            │││││╻           Type
    │   %33 = (Core.tuple)(%2, %17, %32)::Tuple{Axis{:row,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}},Axis{:col,Base.OneTo{Int64}},Axis{:page,Base.OneTo{Int64}}}
    │   %34 = invoke AxisArrays._default_axes(_2::Base.ReshapedArray{Int64,4,UnitRange{Int64},Tuple{}}, ()::Tuple{}, %33::Tuple{Axis{:row,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}},Axis{:col,Base.OneTo{Int64}},Axis{:page,Base.OneTo{Int64}}})::NTuple{4,Axis}
    └──       return %34 
```

it works if you only reshape to only 3 dimensions

```
julia> @code_warntype AxisArrays.default_axes(reshape(1:16, 2,2,4), (.5:.5:1,))
Body::Tuple{Axis{:row,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}},Axis{:col,Base.OneTo{Int64}},Axis{:page,Base.OneTo{Int64}}}
172 1 ─ %1  = (Base.getfield)(args, 1, true)::StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}    │╻╷         _default_axes
    │   %2  = %new(Axis{:row,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}, %1)::Axis{:row,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}}
    │   %3  = (Base.getfield)(A, :dims)::Tuple{Int64,Int64,Int64}                                                                │││╻╷╷╷       axes
    │   %4  = (Base.getfield)(%3, 1, true)::Int64                                                                                ││││╻          axes
    │   %5  = (Base.slt_int)(%4, 0)::Bool                                                                                        │││││╻╷╷        map

    [...]

    │   %25 = %new(Base.OneTo{Int64}, %24)::Base.OneTo{Int64}                                                                    │││││││││  
    │   %26 = %new(Axis{:page,Base.OneTo{Int64}}, %25)::Axis{:page,Base.OneTo{Int64}}                                            │││││╻          Type
    │   %27 = (Core.tuple)(%2, %14, %26)::Tuple{Axis{:row,StepRangeLen{Float64,Base.TwicePrecision{Float64},Base.TwicePrecision{Float64}}},Axis{:col,Base.OneTo{Int64}},Axis{:page,Base.OneTo{Int64}}}
    └──       return %27
```